notebooks/Gradient Boosted Regression Trees - Single Target.ipynb

```diff
@@ -0,0 +1,646 @@
+   },
+   "outputs": [],
+   "source": [
+    "import sklearn.preprocessing\n",
+    "import sklearn.decomposition\n",
+    "import sklearn.ensemble\n",
+    "import sklearn.model_selection\n",
+    "import sklearn.feature_selection\n",
+    "import sklearn.metrics\n",
+    "\n",
+    "from scipy.stats import randint as sp_randint\n",
+    "from scipy.stats import uniform as sp_uniform\n",
+    "\n",
+    "from time import time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def report(results, n_top=3):\n",
+    "    for i in range(1, n_top + 1):\n",
+    "        candidates = np.flatnonzero(results['rank_test_score'] == i)\n",
+    "        for candidate in candidates:\n",
+    "            print(\"Model with rank: {0}\".format(i))\n",
+    "            print(\"Mean validation score: {0:.3f} (std: {1:.3f})\".format(\n",
+    "                  results['mean_test_score'][candidate],\n",
+    "                  results['std_test_score'][candidate]))\n",
+    "            print(\"Parameters: {0}\".format(results['params'][candidate]))\n",
+    "            print(\"\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Gradient boosting regression in skl only supports single (1-d) targets.\n",
+    "Choose the single target you'd like to explore here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "#output_columns = [\"Ca\",\"P\",\"pH\",\"SOC\",\"Sand\"]\n",
+    "output_columns=[\"Ca\"]\n",
+    "\n",
+    "X,targets=load.load_training_spectra()\n",
+    "\n",
+    "y=targets[output_columns]\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Gradient Boosted Trees\n",
+    "Gradient boosted trees can be regarded as the smarter older sibling to the [Random Forest](http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html#sklearn.ensemble.RandomForestRegressor) ensemble method.\n",
+    "Gradient Boosted Trees are iteratively corrected, with the residuals of the previous tree \n",
+    "providing the basis for building the subsequent tree.  \n",
+    "The user specifies the loss function to be optimized,\n",
+    "and here we use least squares regression,\n",
+    "the default loss function for scikit-learn's [GradientBoostingRegressor](http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html).\n",
+    "More info and intuition about gradient boosted trees can be found [here](http://machinelearningmastery.com/gentle-introduction-gradient-boosting-algorithm-machine-learning/)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To help speed up the task of training these trees, \n",
+    "we perform some dimensionality reduction \n",
```

Comment:
dummy links, good call - I wasn't sure about the final notebook name structure, but I've linked other good references that hopefully I'll remember to fill the correct names in for later.